client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts

```diff
@@ -0,0 +1,104 @@
+                        phoneNumber: contact.personByPersonInfo.phoneNumber,
+                        identificationType: contact.personByPersonInfo.identificationType ? contact.personByPersonInfo.identificationType : IdentificationTypes.ID,
+                        identificationNumber: contact.personByPersonInfo.identificationNumber,
+                        birthDate: contact.personByPersonInfo.birthDate,
+                        additionalPhoneNumber: contact.personByPersonInfo.additionalPhoneNumber,
+                        gender: contact.personByPersonInfo.gender,
+                        contactDate: contact.contactEventByContactEvent.startTime,
+                        contactType: contact.contactType,
+                        cantReachContact: contact.cantReachContact ? contact.cantReachContact : false,
+                        extraInfo: contact.extraInfo,
+                        relationship: contact.relationship,
+                        familyRelationship: contact.familyRelationship,
+                        contactedPersonCity: contact.contactedPersonCity,
+                        occupation: contact.occupation,
+                        doesFeelGood: contact.doesFeelGood ? contact.doesFeelGood : false,
+                        doesHaveBackgroundDiseases: contact.doesHaveBackgroundDiseases ? contact.doesHaveBackgroundDiseases : false,
+                        doesLiveWithConfirmed: contact.doesLiveWithConfirmed ? contact.doesLiveWithConfirmed : false,
+                        doesNeedHelpInIsolation: contact.doesNeedHelpInIsolation ? contact.doesNeedHelpInIsolation : false,
+                        repeatingOccuranceWithConfirmed: contact.repeatingOccuranceWithConfirmed ? contact.repeatingOccuranceWithConfirmed : false,
+                        doesWorkWithCrowd: contact.doesWorkWithCrowd ? contact.doesWorkWithCrowd : false,
+                        expand: false
+                    }
+                )
+            });
+        });
+    };
+
+    const updateInteractedContact = (interactedContact: InteractedContact, fieldToUpdate: InteractedContactFields, value: any) => {
+        setCurrentInteractedContact(interactedContact);
+        let contactIndex = interactedContactsState.interactedContacts.findIndex(contact => contact.id === interactedContact.id)
+        interactedContactsState.interactedContacts[contactIndex] = { ...interactedContactsState.interactedContacts[contactIndex], [fieldToUpdate]: value };
+    };
+
+    const changeIdentificationType = (interactedContact: InteractedContact, booleanValue: boolean) => {
+        const newIdentificationType = booleanValue ? IdentificationTypes.PASSPORT : IdentificationTypes.ID;
+        updateInteractedContact(interactedContact, InteractedContactFields.IDENTIFICATION_TYPE, newIdentificationType);
+    };
+
+    const openAccordion = (interactedContact: InteractedContact) => {
+        updateInteractedContact(interactedContact, InteractedContactFields.CANT_REACH_CONTACT, false);
```

Comment:
I think its better if you change both context field in one transaction

Reply:
why